src/transformStateWithClones.js

```diff
@@ -1,13 +1,57 @@
 'use strict';
 
-/**
+/*
  * @param {Object} state
  * @param {Object[]} actions
  *
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const result = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+    const typeAdd = 'addProperties';
```

Comment:
Move 'typeAdd', 'typeRemove', and 'typeClear' variable declarations outside the function to avoid redeclaring them on every function call.